dashboard/pages/api/metric.ts

```diff
@@ -17,17 +17,149 @@
 import { Metrics, MetricsSample } from "../../components/metrics"
 import api from "./api"
 
+export const INTERVAL = 5000
 export interface BackPressuresMetrics {
   outputBufferBlockingDuration: Metrics[]
 }
 
+// Get back pressure from meta node -> prometheus
 export async function getActorBackPressures() {
+  console.log("send api")
   const res: BackPressuresMetrics = await api.get(
     "/metrics/actor/back_pressures"
   )
   return res
 }
 
+export interface BackPressureInfo {
+  actorId: number
+  fragmentId: number
+  downstreamFragmentId: number
+  value: number
+}
+
+export interface BackPressureRateInfo {
+  actorId: number
+  fragmentId: number
+  downstreamFragmentId: number
+  backPressureRate: number
+}
+
+function convertToMapAndAgg(
+  back_pressures: BackPressureInfo[]
+): Map<string, number> {
+  // FragmentId-downstreamFragmentId, total value
+  const map_value = new Map<string, number>()
+  // FragmentId-downstreamFragmentId, total count
+  const map_number = new Map<string, number>()
+  // FragmentId-downstreamFragmentId, average value
+  const map = new Map<string, number>()
+  for (const item of back_pressures) {
+    const key = `${item.fragmentId}-${item.downstreamFragmentId}`
+    if (map_value.has(key) && map_number.has(key)) {
+      // add || tp avoid NaN and pass check
+      map_value.set(key, (map_value.get(key) || 0) + item.value)
+      map_number.set(key, (map_number.get(key) || 0) + 1)
+    } else {
+      map_value.set(key, item.value)
+      map_number.set(key, 1)
+    }
+  }
+
+  for (const [key, value] of map_value) {
+    // add || 1 to avoid NaN and pass check
```

Comment:
`|| 1` doesn't make sense... Also, it seems impossible to get a null or zero from `map_number` here.

Reply:
The reason why I do this is to pass the check: https://github.com/risingwavelabs/risingwave/actions/runs/8009389265/job/21877909103?pr=14863

If not added, the check will always report the error:
```
./pages/api/metric.ts:71:26
Type error: Object is possibly 'undefined'.

  69 |   for (const [key, value] of mapValue) {
  70 |     // add || 1 to avoid NaN and pass check
> 71 |     map.set(key, value / mapNumber.get(key))
     |                          ^
  72 |   }
  73 |   return map
  74 | }
Error: Process completed with exit code 1.
```

Because here is divide, so I use the `xx || 1` rather than `xx || 0`